src/uas/UAS.cc

```diff
@@ -2410,56 +2410,198 @@ void UAS::setManualControlCommands(float roll, float pitch, float yaw, float thr
+                    0,
+                    0,
+                    fx,
+                    fy,
+                    fz,
+                    0,
+                    0
+                    );
+        } else if (joystickMode == JoystickInput::JOYSTICK_MODE_VELOCITY) {
+            // Send the the local velocity setpoint (local pos sp external message)
+            static float vx = 0;
+            static float vy = 0;
+            static float vz = 0;
+            static float yawrate = 0;
+            //XXX: find decent scaling
+            vx -= pitch;
+            vy += roll;
+            vz -= 2.0f*(thrust-0.5);
+            yawrate += yaw; //XXX: not sure what scale to apply here
+            uint16_t typeMask = (1<<10)|(7<<6)|(7<<0); // select only VELOCITY control
+            mavlink_msg_set_position_target_local_ned_pack(mavlink->getSystemId(),
+                    mavlink->getComponentId(),
+                    &message,
+                    QGC::groundTimeUsecs(),
+                    this->uasId,
+                    0,
+                    MAV_FRAME_LOCAL_NED,
+                    typeMask,
+                    0,
+                    0,
+                    0,
+                    vx,
+                    vy,
+                    vz,
+                    0,
+                    0,
+                    0,
+                    0,
+                    yawrate
+                    );
+        } else if (joystickMode == JoystickInput::JOYSTICK_MODE_MANUAL) {
+            if(((base_mode & MAV_MODE_FLAG_DECODE_POSITION_MANUAL) && (base_mode & MAV_MODE_FLAG_DECODE_POSITION_SAFETY)) || (base_mode & MAV_MODE_FLAG_HIL_ENABLED)) {
+
+                // Transmit the manual commands only if they've changed OR if it's been a little bit since they were last transmit. To make sure there aren't issues with
+                // response rate, we make sure that a message is transmit when the commands have changed, then one more time, and then switch to the lower transmission rate
+                // if no command inputs have changed.
+                // The default transmission rate is 50Hz, but when no inputs have changed it drops down to 5Hz.
+                bool sendCommand = false;
+                if (countSinceLastTransmission++ >= 10) {
+                    sendCommand = true;
+                    countSinceLastTransmission = 0;
+                } else if ((!isnan(roll) && roll != manualRollAngle) || (!isnan(pitch) && pitch != manualPitchAngle) ||
+                        (!isnan(yaw) && yaw != manualYawAngle) || (!isnan(thrust) && thrust != manualThrust) ||
+                        buttons != manualButtons) {
+                    sendCommand = true;
+
+                    // Ensure that another message will be sent the next time this function is called
+                    countSinceLastTransmission = 10;
+                }
+
+                // Now if we should trigger an update, let's do that
+                if (sendCommand) {

```

Comment:
No, because the other modes send direct set points to the vehicle, which has an internal timeout. The joystick is acting like the high-level companion computer in these modes, and much higher updates rates are expected by the safety timeout logic in those cases.